src/common/signal/signal_action.h

```diff
@@ -71,6 +71,23 @@ class SignalAction : public NotCopyable {
    * Additionally, two guard pages will be allocated to bookend the usable area.
    */
   const size_t altstack_size_;
+  /**
+   * Determine the number of bytes to allocate to altstack_size_.
+   * Makes sure that it is a multiple of the system page size.
+   */
+  size_t DetermineAltStackSize() const {
+    auto min_stack_size = static_cast<size_t>(MINSIGSTKSZ);
+
+    if (guard_size_ * 4 > min_stack_size) {
+      return guard_size_ * 4;
+    }
+
+    if (min_stack_size % guard_size_ != 0) {
+      min_stack_size += guard_size_ - (min_stack_size % guard_size_);
+    }
+
+    return min_stack_size;
+  }
```

Comment:
Nice work in tracking this down!

If we use page size instead of guard size, then it becomes clear that we are aligning to the page size.

Perhaps something along the lines of:
```
const size_t min_alt_stack_size = 4 * page_size;
const size_t stack_size_page_count = IntRoundUpDivide(min_stack_size, page_size);
const size_t stack_size_bytes = page_size * stack_size_page_count;
return std::max(stack_size_bytes, min_alt_stack_size);
```

I'm not completely in sync with the names of the variables above. If possible, can we find better names that capture the semantics of the values we calculate (e.g. there is min stack size and also 4x of that ... do we have a semantic distinction in these concepts?).
```

Reply:
My understanding from the original imp was that `altstack_` should always be at least 4*page size, but if `MINSIGSTKSZ` is greater we default to that, so I kept the same condition to choose between the two in the new logic